clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/security/KubernetesApiClientConfig.groovy

```
@@ -22,18 +22,65 @@ import io.kubernetes.client.util.KubeConfig;
 import io.kubernetes.client.util.SSLUtils;
 
 import groovy.util.logging.Slf4j
+import org.apache.commons.lang3.StringUtils
+import org.yaml.snakeyaml.Yaml
+import org.yaml.snakeyaml.constructor.SafeConstructor
 
 import javax.net.ssl.KeyManager;
 
 @Slf4j
 public class KubernetesApiClientConfig extends Config {
   String kubeconfigFile
+  String context
+  String cluster
+  String user
+  String userAgent
 
-  public KubernetesApiClientConfig(String kubeconfigFile) {
+  public KubernetesApiClientConfig(String kubeconfigFile, String context, String cluster, String user, String userAgent) {
     this.kubeconfigFile = kubeconfigFile
+    this.context = context
+    this.user = user
+    this.userAgent = userAgent
   }
 
   public ApiClient getApiCient() throws Exception {
-    return (kubeconfigFile ? fromConfig(kubeconfigFile) : Config.defaultClient())
+    KubeConfig kubeconfig
+
+    try {
+      if (StringUtils.isEmpty(kubeconfigFile)) {
+        kubeconfig = KubeConfig.loadDefaultKubeConfig()
+      } else {
+        kubeconfig = KubeConfig.loadKubeConfig(new FileReader(kubeconfigFile))
+      }
+    } catch (FileNotFoundException e) {
+      throw new RuntimeException("Unable to create credentials from kubeconfig file: " + e, e)
+    } catch (Exception e2) {
```

You should probably handle this as well ( you can just throw an exception )